modules/build/src/main/scala/scala/build/bsp/BspClient.scala

```diff
@@ -229,9 +231,16 @@ class BspClient(
           new b.Diagnostic(range, diag.message)
 
         diag.textEdit.foreach { textEdit =>
-          val bTextEdit = TextEdit(range, textEdit.newText)
-          bDiag.setData(bTextEdit.toJsonTree())
+          val bScalaTextEdit      = new ScalaTextEdit(range, textEdit.newText)
+          val bScalaWorkspaceEdit = new ScalaWorkspaceEdit(List(bScalaTextEdit).asJava)
+          val bAction             = new ScalaAction(textEdit.title)
+          bAction.setEdit(bScalaWorkspaceEdit)
+          val bScalaDiagnostic = new ScalaDiagnostic
+          bScalaDiagnostic.setActions(List(bAction).asJava)
+          bDiag.setData("scala")
```

Comment:
```suggestion
```

Reply:
oooo dang, this is a good catch!!! This should be `setDataKind`. Thanks for catching this!